stdlib/public/Cxx/CxxStack.swift

```diff
@@ -0,0 +1,59 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// A C++ type that represents a stack.
+///
+/// C++ standard library types `std::stack` conforms to this protocol.
+public protocol CxxStack<Element> {
+  associatedtype Element
+  associatedtype Size: BinaryInteger
+  associatedtype RawReference: UnsafeCxxInputIterator
+    where RawReference.Pointee == Element
+
+  /// Do not implement this function manually in Swift.
```

Comment:
I don't think you need this comment for anything that you expect the user to call. It would make sense for something like `__topUnsafe()`  but not so much for a method you expect the user to call like `empty()`.